### ISSUE TITLE
Include the packages version in the release PR title if they are all the same

### ISSUE THREAD

**Issue by Veetaha** [MarcoIeni/release-plz#1582]
## Motivations

Hi! I'm using release-plz for [this project](https://github.com/elastio/bon). I'm very satisfied with how it works so far. One small nit that I have is that `release-plz` creates a release PR with a very generic short title that I'd like to customize. This is because I use squash-and-merge strategy for PRs, so the PR title becomes inherited by the commit message header. The commit then looks like this:
![image](https://github.com/user-attachments/assets/928b8a76-6792-4d33-a58e-81e63ea14cce).

I'd like, for example, the commit message (and thus the PR title, which is its origin) to mention the version of the release like this: `chore: release v1.0.5`. This most likely requires some templating for the PR title config.

## Solution

Basically, it could be a new config key in `release-plz` `[workspace]` stanza.

The challenge with version templating will be with the cases when there are many different crates released with different versions. I suppose release-plz should just return an error in this case if the crates in the cargo workspace don't use a single version.

## Additional context

The lack of this feature isn't a big deal for me, it's mostly an esthetic nit that I'd like to visually see groups of commits from different versions separated by such release commits when browsing

**Comment by MarcoIeni** [comment]
Great suggestion! Release-plz already puts the version in the pr title for single packages (not workspaces).

However we could add the version to the title also if a workspace version is present.

Do you want to work on this? 👀

**Comment by Veetaha** [comment]
I'd suggest adding a version to the PR title disregarding if workspace inheritance is used or not, because workspace inheritance doesn't mean that all packages **must** inherit the version. Some packages in the workspace are free to ignore workspace inheritance and override their own version.

I'd propose that the PR title should include the version if all the released packages have the same version (no matter if workspace inheritance is used).

What do you think? 

Yeah, I can implement that change in a PR to release-plz. Though I won't promise it'll be fast, but the impl should be simple according to the code I've seen

**Comment by MarcoIeni** [comment]
I agree with you!

PR welcome. No rush!

**Comment by MarcoIeni** [comment]
I assigned this issue to you just to organize things better on my end 👍 
If you don't have time to work on this, let me know at any time 🙏
